### ISSUE TITLE
Position Error

### ISSUE THREAD

**Issue by Patschke** [FormerLurker/Octolapse#131]
#### Version of Octolapse
 
Octolapse Version: 0.3.1

#### Version of OctoPrint

OctoPrint Version: 1.3.7

#### When you ran into the problem, did you have diagnostic logging enabled?

Diagnostic Logging was Enabled: no

#### What were you doing when the problem occurred

1. Start a print with active Octolapse
2. Pause the print
3. Manually move the extruder out of print area (using control tab buttons)
4. Change filament
5. Resume print
6. Extruder moves itself back to position it was when I paused and print continues

#### What should have happened?
Octolaps ignores whatever I desire to do while the print is paused, and continue operation after it continues. 

#### What happened instead?
Octolapse gave me an error as the extruder left the print area: 
"Position - Coordinates (X:-4.83900,Y:94.30800,Z:2.43400,E:343.23320) are out of the printer area! Cannot resume position tracking until the axis is homed, or until absolute coordinates are received."

#### Operating System running OctoPrint and Octolapse

OS Name: Raspian
Os Version: Jessie

#### Printer model & used firmware incl. version

Printer Model: Anet A8
Printer Firmware Version: stock

#### Browser and version of browser, operating system running browser

Browser: Firefox
Browser OS: Windows 7

#### TL;DR ####
Octolapse should not care what happens while the print is paused. It should ignore coordinates out of print area (e.g. negative coordinates) in that case.

**Comment by FormerLurker** [comment]
You could just increase your print volume within either the Ocotoprint printer profile (settings->Printer Profiles-> Click Edit next to your profile-> Print bed and build volume -> Custom bounding box) or you could do this within Octolapse (Settings-> Octolapse-> Printer tab-> Click edit next to your printer-> Override Octoprint Printer Profile).

I could suppress tracking when the print is paused, but then Octolapse might take a snapshot in the wrong place (or worse yet return to the incorrect coordinates) if things aren't perfectly realigned when you resume the print.

I'll think about this and see if there is a way to safely ignore position updates when the printer is paused, or to at least suppress the error messages during that time.  However, it would be good if you updated your print volume (looks like X goes to at least -5.0).  My printer primes at y=-3, and if this isn't entered into the print volume Octolapse will complain.  What happens behind the scenes is when Octolapse is out of bounds is won't take another snapshot until it receives absolute coordinates or a position update.  For some prints containing only relative XYZ or relative E coordinates, an out of bounds error might mean no more snapshots for the remainder of the print since Octolapse can't know exactly where the printhead is.  This was added as a safety measure, but it is a little overbearing.  Perhaps I could add an option to suppress the position warning messages completely?  I have also been thinking of adding a print volume suggestion to the error message by building up all of the out of bounds coordinates and expanding the volume to match.  Maybe something like this, "Octolapse has detected that your print volume is X,Y,Z.  Do you want to update your printer profile to match?"

Let me know if changing your volume works and I'll keep pondering a way to make this more user friendly.

Thanks for submitting an issue!

**Comment by jowenteo** [comment]
hi all, 
i have the same issue with prusa mk2s . 
everything was default and in test mode , give me error " Octolapse - Position Error
Position - Coordinates (X:12.00000,Y:-2.00000,Z:0.20000,E:20.70000) are out of the printer area! Cannot resume position tracking until the axis is homed, or until absolute coordinates are received." 
when i started a test print . 
care to advise ? tia

**Comment by FormerLurker** [comment]
Sure!  You have two options:

1.  Adjust your print volume within Octoprint here:  Settings->Printer Profiles->Print bed and build volume
2.  Override the print volume within Octolapse here: Settings->Octolapse->Printers->Override Octoprint Printer Profile)

Here is what I use for my Mk2:

X:  0 - 250
Y:  -3 - 213 (some room for play on the upper end)
Z:  0 - 207 (some people go higher, I like to be safe)

I have seen gcode where the X coordinate goes negative too, but never in mine so I leave Xmin = 0.

Do you all think I should cook these settings into the default printer profiles, or should I keep it as is and use the Octoprint printer volume settings by default?  I suppose I could do both, but I'm conflicted...

**Comment by jowenteo** [comment]
let me try

**Comment by jowenteo** [comment]
![screen shot 2018-04-11 at 12 01 02 am](https://user-images.githubusercontent.com/38254389/38568528-75dedf66-3d1b-11e8-8ae0-869a1f2eb1a8.png)
where should i input the numbers?

**Comment by FormerLurker** [comment]
You are using -213 and -207 for your max, but it should be 213 and 207 (sorry, I used a hyphen in my comment above when I shouldn't have since it was confusing!)

**Comment by jowenteo** [comment]
![screen shot 2018-04-11 at 12 27 39 am](https://user-images.githubusercontent.com/38254389/38569944-32cb2118-3d1f-11e8-8ce7-c1ae96347399.png)
for rendering , which one is a better options ? 
thanks so much for the prompt response .

**Comment by jowenteo** [comment]
> Sure! You have two options:

Adjust your print volume within Octoprint here: Settings->Printer Profiles->Print bed and build volume
Override the print volume within Octolapse here: Settings->Octolapse->Printers->Override Octoprint Printer Profile)
Here is what I use for my Mk2:

X: 0 - 250
Y: -3 - 213 (some room for play on the upper end)
Z: 0 - 207 (some people go higher, I like to be safe)

it works now . no more error msg ! thank you so much . hope to see updates from you soon

**Comment by FormerLurker** [comment]
It depends.  I like to use the Fixed Length 5s profile.  If there are lots of frames (very tall print, timer trigger, etc) you may want to use something longer.  Otherwise I recommend starting with 30FPS.

I think after you create a few timelapses you'll start to figure out what effect the fixed length/fixed fps does depending on the number of frames captured.  It's best to play around with the settings to get a feel for what's happening.

**Comment by FormerLurker** [comment]
I deem this issue closed!  Let me know if you have any more problems.

**Comment by edasque** [comment]
This is happening to me again since the last release of Octoprint (RC and final of 1.3.10). I don't think anything else has changed.

**Comment by FormerLurker** [comment]
Erik, perhaps your Octoprint printer profile was changed/migrated during the upgrade?  In any case, to fix it for good, open your octolapse printer profile, select 'override Octoprint printer volume' (the name may be a bit different, not at my PC to check), and enter your exact printer volume there.  If you still get position errors, pay attention to the exact error message.  It will tell you the out of bounds location and will help you adjust the volume.  Please note that some printers can have negative coordinates!
